src/cardano_common_plug.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Common code for cracking Cardano 128-byte length legacy secret Keys (a.k.a XPrv).
+ *
+ * This software is Copyright (c) 2022, Pal Dorogi <pal dot dorogi at gmail.com>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * This file takes replicated but common code, shared between the CPU
+ * and the GPU formats, and places it into one common location.
+ *
+ * This files is common for its future use, and currently there isn't an OpenCL format
+ * implementing this feature.
+ *
+ */
+
+#include <stdio.h>
+#include "arch.h"
+#include "cardano_common.h"
+#include "jumbo.h"
+
+struct fmt_tests cardano_tests[] = {
+	//
+	// https://github.com/CardanoSolutions/ByronWalletRecovery/blob/63ac66ee3532b2a1c72a88c0aa726d0f1ce3ed7a/tools/inspect-keystore/index.js#L34
+	// XPrv is BIP32-Ed25519 key, derived from an extended Ed25519 key (64-byte sk + 32-byte pk)
+	// by appending a 32-byte length chaincode to it, and clear and extra bit ( esk[31] &= 0x20).
+	//      +---------------------------------+-----------------------+-----------------------+
+	//      | Extended Private Key (64 bytes) | Public Key (32 bytes) | Chain Code (32 bytes) |
+	//      +---------------------------------+-----------------------+-----------------------+
+	//      <------------ ENCRYPTED ---------->
+	{ "$cardano$1$b57361ebe335fa171a260fea7d3277579c212dc74fc2a408d6cbd8a6e7a847cab3c44c5fb190705ddd2698f2d5390798893349b4321e7474b1ce06c9d410b3d6055b42d4a95f19cb34b516a160a306c0eaef398e70ea91da450ccb2a7819e95b0102030405060708091011121314151617181920212223242526272829303132", "Secret1234" },
+	{ NULL }
+};
+
+int cardano_valid(char *ciphertext, struct fmt_main *self)
+{
+	char *ctcopy, *keeptr, *p;
+	int len;
+
+	if (strncmp(ciphertext, FORMAT_TAG, FORMAT_TAG_LEN) != 0)
+		return 0;
+
+	ctcopy = xstrdup(ciphertext);
+	keeptr = ctcopy;
+
+	ctcopy += FORMAT_TAG_LEN;
+	if ((p = strtokm(ctcopy, "$")) == NULL) // scheme
+		goto err;
+	if (!isdec(p))
+		goto err;
```

Comment:
If we currently only support `1` there, we should insist on that - so e.g. instead of `!isdec(p)`, we can `strcmp(p, "1")`.

Also, we don't have to decode that field into the salt structure until we use it, nor to have a field in there. We can just have a comment on its possible future use, here in `valid`.

Reply:
Fixed